packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm

```diff
@@ -339,6 +341,10 @@ - (void)_start
     if (RCTGetUseNativeViewConfigsInBridgelessMode()) {
       installLegacyUIManagerConstantsProviderBinding(runtime);
     }
+    
+    RCTExecuteOnMainQueue(^{
+      RCTRegisterReloadCommandListener(self);
```

Comment:
I'm not sure if this reload change is necessary! 

But, if it is, I have a request: Can we create a custom implementation of RCTReloadListener that retains RCTInstance weakly?

Bridgeless mode adheres to this principle: 
> Nothing outside of the RCTHost should own a reference to RCTInstance.

In practice, this line is fine: RCTRegisterReloadCommandListener holds on to RCTReloadListeners weakly. 

But, I think bridgeless mode should not depend on this guarantee:
1. Weak retention of listeners is an implementation detail of RCTRegisterReloadCommandListener. If someone changes it in the future, this bridgeless principle could be violated.
2. Bridgeless classes should be responsible for enforcing bridgeless principles.

Reply:
This was needed to make `Reload` button work in Redbox. I understand your concerns, I've tweaked the code to listen to notification through the notification center so now we don't need to worry about retaining this instance. WDYT about this approach?

Reply:
@okwasniewski, the current solution definitely addresses the instance retention issue.

But, I don't think it's a good idea to rely on a global event emission to communicate between RCTRedBox and RCTInstance. Many apps/environments actually run multiple react instances at the same time. Global event emission makes that configuration on iOS more difficult. And also, more generally: if we can avoid reliance on global things, we probably should. 

Was there any problem with creating a custom RCTReloadListener? Perhaps you noticed an issue, or have a concern, that hadn't crossed my mind.